components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -163,6 +193,17 @@ func (r *NotebookReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		}
 	}
 
+	events, err := r.eventLister.Events(req.Namespace).List(labels.Everything())
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	for _, event := range events {
```

Comment:
What if the failure event is not exist yet at this point?

Reply:
Should be detected in a later call to `Reconcile` right? The failure event will trigger a reconciliation for sure (we watch all events from sts and pods). And at that point, both the nb instance and the ss will exist, so that statements will be always executed.

Reply:
This is also solved in https://github.com/kubeflow/kubeflow/commit/d032910ce8cee317e09193a108fea4c785110aa1, since the event is the first object that we fetch in the reconcile call.